DynatraceAppMon/Context.cs

```diff
@@ -250,6 +254,89 @@ internal bool goToMethod(Lookup lookup)
             }
         }
 
+        private bool HasClassFiles(ProjectItem projectItem)
+        {
+            bool classFileOccurred = false;
+            for (int i = 1; i <= projectItem.FileCount; i++)
+            {
+                string fileExtension = Path.GetExtension(projectItem.FileNames[(short)i]);
+                if (fileExtension != null && (fileExtension.Contains(".cs") || fileExtension.Contains(".vb")))
+                {
+                    classFileOccurred = true;
+                    break;
+                }
+            }
+            return classFileOccurred;
+        }
+
+        private List<String> GetClassFiles(ProjectItem projectItem)
+        {
+            List<String> classFiles = new List<string>();
+            for (int i = 1; i <= projectItem.FileCount; i++)
+            {
+                string fileName = projectItem.FileNames[(short)i];
+                string fileExtension = Path.GetExtension(fileName);
+                if (fileExtension != null && (fileExtension.Contains(".cs") || fileExtension.Contains(".vb")))
```

Comment:
HasClassFiles and GetClassFiles contains lots of same logic, this logic should be common